### ISSUE TITLE
Logger - DS2 LTV fails if adaption response returns NACK

### ISSUE THREAD

**Issue by dschultzca** [RomRaider/RomRaider#107]
Handle adaption NACK response better.
https://www.romraider.com/forum/viewtopic.php?p=158155#p158155

**Comment by GTO2013** [comment]
I looked into this today and the problem is that the sendRcv function in DS2LoggerConnection gets a predefined length it expects as a response. The SerialManager will wait until the expected number of bytes arrived, if they don't, it will give a read timeout and discard the read data. A trivial fix would be to still put the read bytes into the response array, but that might cause some unexpected behavior in other methods. Because the data is not copied into the response array, the DS2ResponseProcessor gives back the wrong error in the validateResponse function. 

I also implemented  DS2 interfacing code some time ago and the way I read the response was to first only wait for three bytes: Ecu address, length, ack. I would then read the remaining data which was specified as length. It is done the same way in the BMW toolset like INPA. It might be better to only work with response arrays that don't expect a certain length but are always the size of the actual response without zero bytes. So you reserve memory based on the returned length byte instead of using the constructXXXResponse methods in DS2LoggerProtocol.java.
